File: SDKs/shutterstock.com/1.1.5/python/src/sdk/models/shared/instrumentslistresult.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import instrument as shared_instrument


@dataclass_json
@dataclasses.dataclass
class InstrumentsListResult:
    r"""InstrumentsListResult
    Image search results
    """
    
    data: Optional[list[shared_instrument.Instrument]] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    page: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('page') }})
    per_page: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('per_page') }})
    

File: SDKs/shutterstock.com/1.1.5/typescript/src/sdk/models/shared/instrumentslistresult.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { Instrument } from "./instrument";



// InstrumentsListResult
/** 
 * Image search results
**/
export class InstrumentsListResult extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data", elemType: Instrument })
  data?: Instrument[];

  @SpeakeasyMetadata({ data: "json, name=page" })
  page?: number;

  @SpeakeasyMetadata({ data: "json, name=per_page" })
  perPage?: number;
}
